src/App.tsx

```diff
@@ -1,26 +1,176 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import * as todosService from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { Header } from './components/header';
+import { Footer } from './components/footer';
+import { TodoList } from './components/TodoList';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [id, setId] = useState(0);
 
-const USER_ID = 0;
+  enum Status {
+    Active = 'active',
+    Completed = 'completed',
+    All = 'all',
+  }
+  if (error !== '') {
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  }
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  function handleToggle(todoId: number) {
+    setLoading(true);
+    setId(todoId);
+
+    todosService
+      .patchTodo(todos.find(todo => todo.id === todoId).id, {
+        completed: !todos.find(todo => todo.id === todoId).completed,
+      })
+      .then(() => {
+        setTimeout(() => {
+          setTodos(currentTodos =>
+            currentTodos.map(todo =>
+              todo.id === todoId
+                ? { ...todo, completed: !todo.completed }
+                : todo,
+            ),
+          );
+          setLoading(false);
+        }, 500);
+      })
+      .catch(() => setError('Unable to update a todo'));
   }
 
+  function deleteTodo(todoId) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const deleteTodo = (todoId) => {
```